src/main/java/bbm/brickbreaker/NetworkController.java

```diff
@@ -0,0 +1,152 @@
+package bbm.brickbreaker;
+
+import basicneuralnetwork.NeuralNetwork;
+
+import java.awt.*;
+import java.util.*;
+import static bbm.brickbreaker.Bounds.*;
+
+public class NetworkController {
+    private final Ball ball;
+    private final Paddle paddle;
+    private final int radius;
+    private static final int NUM_GENERATIONS = 50;
+    private static final int GENERATION_SIZE = 1000;
+    private ArrayList<NeuralNetwork> newGeneration = new ArrayList<>();
+    private ArrayList<NeuralNetwork> top10 = new ArrayList<>();
+
+    public NetworkController(Ball ball, int radius, Paddle paddle) {
+        this.ball = ball;
+        this.paddle = paddle;
+        this.radius = radius;
+    }
+
+    public void generate() {
+        for (int i = 0; i < GENERATION_SIZE; i++) {
+            NeuralNetwork network = new NeuralNetwork(2, 4, 2);
+            newGeneration.add(network);
+        }
+
+        for (int i = 1; i < NUM_GENERATIONS; i++) {
+            System.out.println("Curr gen: " + i);
+            getTop10();
+            updateNetwork();
+        }
+    }
+
+    public void updateNetwork() {
+        newGeneration = new ArrayList<>();
+        for (int i = 0; i < GENERATION_SIZE; i++) {
+            NeuralNetwork child = null;
+            for (int j = 0; j < 10; j++) {
+                NeuralNetwork parent1 = top10.get(j);
+                if (j == 0) {
+                    NeuralNetwork parent2 = top10.get(j++);
+                    child = parent1.merge(parent2);
+                } else {
+                    child = child.merge(parent1);
+                }
+            }
+
+            child.mutate(0.1);
+            newGeneration.add(child);
+        }
+    }
+
+    public void getTop10() {
+        HashMap<NeuralNetwork, Integer> genTop10 = new HashMap<>();
+
+        for (NeuralNetwork neuralNetwork : newGeneration) {
+            Network network = new Network(neuralNetwork, paddle, ball);
```

Comment:
Keep score in here too.
It will make your life easier because you won't need a map